src/crimson/os/alienstore/alien_store.cc

```diff
@@ -619,9 +619,12 @@ std::vector<uint64_t> AlienStore::_parse_cpu_cores()
     std::string cpu;
     std::getline(cpu_stream, cpu, '-');
     uint64_t start_cpu = std::stoull(cpu);
+    uint64_t end_cpu = start_cpu;
     std::getline(cpu_stream, cpu, '-');
-    uint64_t end_cpu = std::stoull(cpu);
-    for (uint64_t i = start_cpu; i < end_cpu; i++) {
+    if (!cpu.empty()) {
+      end_cpu = std::stoull(cpu);
+    }
+    for (uint64_t i = start_cpu; i <= end_cpu; i++) {
```

Comment:
i'd suggest use `seastar::resource::cpuset`. will come up with a patch shortly.

Reply:
#48716

Reply:
> #48716

Well. I think this is a better implementation.

Reply:
@cyx1231st We can merge this without running the teuthology suite.  The required lab quay instance is down and we haven't actually started requiring it yet, so I think we'll just delay that until the lab is back up.

Reply:
@ljx023 What's the disadvantage to using seastar's parsing code?

Reply:
> @ljx023 What's the disadvantage to using seastar's parsing code?

I mean using seastar's parsing code is a better implementation. There is little difference between the two methods. Using seastar's parsing code does short the path.

Reply:
Oh, I misunderstood.  Agreed!